core/app/models/spree/notification_dispatcher/action_mailer_dispatcher.rb

```diff
@@ -0,0 +1,51 @@
+# Default sender class for Spree::NotificationDispatcher, sends via
+# ActionMailer.
+#
+# You can see the mapping from notification message types to
+# ActionMailer classes and methods in `.mailer_dispatch_table`, which
+# you can also customize to map to different mailers or methods.
+#
+# The particular argument list for each message type is
+# specified in NotificationDispatcher.signatures
+class Spree::NotificationDispatcher::ActionMailerDispatcher
+  class_attribute :mailer_dispatch_table
+  # values are pairs of fully qualified mailer class names, and method names
+  self.mailer_dispatch_table = {
+    carton_shipped: ['Spree::CartonMailer', :shipped_email],
+    order_confirm: ['Spree::OrderMailer', :confirm_email],
+    order_cancel: ['Spree::OrderMailer', :cancel_email],
+    order_inventory_cancel: ['Spree::OrderMailer', :inventory_cancellation_email],
+    reimbursement_processed: ['Spree::ReimbursementMailer', :reimbursement_email]
+  }
+
+  attr_reader :message
+  attr_reader :delivery_method
+
+  def initialize(message)
+    @message = message.to_sym
+  end
+
+  def deliver(*args)
+    action_mailer, method = lookup_dispatch
+    action_mailer.send(method, *args).deliver_later
+  end
+
+  def lookup_dispatch
+    klass_name, method = mailer_dispatch_table[message]
+
+    # Special behavior for deprecated carton_shipped_email_class config
+    if message == :carton_shipped &&
+       Spree::Config.carton_shipped_email_class &&
+       Spree::Config.carton_shipped_email_class.name != klass_name
+      klass = Spree::Config.carton_shipped_email_class
+      ActiveSupport::Deprecation.warn(":carton_shipped_email_class on Spree::Config is deprecated, please customize the Spree::NotificationDispatcher::ActionMailerDispatcher.dispatch_table instead.")
+    end
+
+    unless klass_name.present? && message.present?
+      raise ArgumentError("Spree::NotificationDisaptcher::ActionMailerDispatcher: no dispatch found for message `#{message}`. Do you need to configure Spree::NotificationDispatcher::ActionMailerDispatcher.dispatch_table ?")
```

Comment:
i think there isn't the typo you think, but I see a few other bugs in this passage now that you bring my attention to it, that perhaps are making it confusing what the intent is. I'll clean it up if this is a good direction. Thanks!